openapi3/path_item.go

```diff
@@ -122,10 +122,10 @@ func (pathItem *PathItem) SetOperation(method string, operation *Operation) {
 }
 
 // Validate returns an error if PathItem does not comply with the OpenAPI spec.
-func (pathItem *PathItem) Validate(ctx context.Context) error {
-	for _, operation := range pathItem.Operations() {
+func (pathItem *PathItem) Validate(ctx context.Context, path string) error {
```

Comment:
I appreciate the effort & I think this is worthwhile! However I'd like to not change the prototype of the Validate here.

Can't you just add the path in case of error in the caller to this Validate?
```go
// somewhere in callback.go and paths.go...
		if err := v.Validate(ctx); err != nil {
			return fmt.Errorf("path %s failed validation %v", path, err)
		}

// ...and in here path_item.go:
		if err := operation.Validate(ctx); err != nil {
			return fmt.Errorf("for operation %s: %v", method, err)
		}
```
What do you think? This is probably complexifying things for not much gain actually

Reply:
Yes, indeed. I can wrap the error when it bubbles up to the caller and add the path information there. I'll make the change.

Reply:
@fenollp I've made the change.

Reply:
I just saw https://github.com/getkin/kin-openapi/pull/596 in the latest release. I don't think there is a need to unwrap the inner errors in this case so I think using `%v` is fine here. Let me know what you think.